### ISSUE TITLE
`okta_policy_rule_signon` update in-place of `priority` value works, but errors

### ISSUE THREAD

**Issue by jlj77** [okta/terraform-provider-okta#424]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

`Terraform v0.14.10`
`oktadeveloper/okta v3.10.1`

*(Note: I'm running this from Terraform Cloud, with remote execution mode.)*

### Affected Resource(s)

- `okta_policy_rule_signon`

### Terraform Configuration Files

```
resource "okta_policy_signon" "p_test1b" {
  name            = "Prompt For MFA - Test"
  status          = "ACTIVE"
  description     = "PoC test sign-on policy 1.b.i-ii"
  priority        = 1
  groups_included = [okta_group.g_test.id]
}

resource "okta_policy_rule_signon" "r_test1b1" {
  policyid           = okta_policy_signon.p_test1b.id
  name               = "MFA - Test"
  status             = "ACTIVE"
  priority           = 1
  authtype           = "ANY"
  network_connection = "ANYWHERE"
  mfa_required       = true
  mfa_prompt         = "SESSION"
  mfa_lifetime       = 15
  session_lifetime   = 1440
  users_excluded     = [okta_user.test1.id]
}

resource "okta_policy_rule_signon" "r_test1b3" {
  policyid            = okta_policy_signon.p_test1b.id
  name                = "MFA - Zone D1"
  status              = "ACTIVE"
  priority            = 2
  authtype            = "ANY"
  network_connection  = "ZONE"
  network_includes    = [okta_network_zone.p_test1di.id]
  mfa_required        = true
  mfa_remember_device = true
  mfa_prompt          = "SESSION"
  mfa_lifetime        = 2160
  session_lifetime    = 2880
}

resource "okta_policy_rule_signon" "r_test1b2" {
  policyid           = okta_policy_signon.p_test1b.id
  name               = "Global Deny"
  status             = "ACTIVE"
  priority           = 3
  authtype           = "ANY"
  network_connection = "ANYWHERE"
  access             = "DENY"
  users_excluded     = [okta_user.test1.id]
}
```

### Debug Output

[Gist of plan and apply output, and an excerpt of the resulting state file](https://gist.github.com/jlj77/d7797153e6b252b9ab3454004ec5b804)

### Panic Output

N/A

### Expected Behavior

- Expected Rule *MFA - Zone D1* to be inserted as Priority 2
- Expected Rule *Global Deny* to be updated, from Priority 2 to Priority 3 (i.e., pushed down the list)

### Actual Behavior

- The above did, in fact, happen
  - Okta, and the state file -- see the Gist above -- both reflected the change (correctly, I believe)
- However, Terraform also threw an error on the run
- A subsequent run, where I made a superficial change to `mfa_lifetime` on one of the policies, was successful, suggesting that the state file from the previous run was indeed correct

### Steps to Reproduce

1. Create a sign-on policy
2. Create two sign-on rules under it
3. Assign a priority of 1 to one, and 2 to the other
4. `terraform apply`
5. Create a third sign-on rule, under the above sign-on policy
6. Assign a priority of 2 to this rule
7. Change the priority of the current 'Priority 2 rule' to be 3
8. `terraform apply`

### References

- #0000

**Comment by bogdanprodan-okta** [comment]
Hi, @jlj77! Thanks for submitting this issue! I'll try to reproduce the steps you've provided and get back to you asap.

**Comment by bogdanprodan-okta** [comment]
Hi, @jlj77! The thing about the priority of the rules is that it should be 1,2,3,4... Even if you make rules with priority like 1,2,10,22,123 it will still convert it to 1,2,3,4,5. Basically, in your scenario, you set priority=2 to the new rule, and priority=3 to the existing one. In case terraform updates the existing rule first, and then creates another one, the priority will stay 2 even if you put 3, and then tries to create the rule with the same priority 2. 

tl;dr the sequence of the rules create/update operations affects the priority.

To fix your issue, just do like this

```
resource "okta_policy_rule_signon" "r_test1b1" {
  policyid           = okta_policy_signon.p_test1b.id
  name               = "MFA - Test"
  status             = "ACTIVE"
  priority           = 1
  authtype           = "ANY"
  network_connection = "ANYWHERE"
  mfa_required       = true
  mfa_prompt         = "SESSION"
  mfa_lifetime       = 15
  session_lifetime   = 1440
  users_excluded     = [okta_user.test1.id]
}

resource "okta_policy_rule_signon" "r_test1b3" {
  policyid            = okta_policy_signon.p_test1b.id
  name                = "MFA - Zone D1"
  status              = "ACTIVE"
  priority            = 3
  authtype            = "ANY"
  network_connection  = "ZONE"
  network_includes    = [okta_network_zone.p_test1di.id]
  mfa_required        = true
  mfa_remember_device = true
  mfa_prompt          = "SESSION"
  mfa_lifetime        = 2160
  session_lifetime    = 2880

  depends_on = [
    okta_policy_rule_signon.r_test1b2 // this will force the terraform to create the 3rd rule first and than update the priority in this one
  ]

}

resource "okta_policy_rule_signon" "r_test1b2" {
  policyid           = okta_policy_signon.p_test1b.id
  name               = "Global Deny"
  status             = "ACTIVE"
  priority           = 2
  authtype           = "ANY"
  network_connection = "ANYWHERE"
  access             = "DENY"
  users_excluded     = [okta_user.test1.id]
}
```

**Comment by jlj77** [comment]
Makes sense -- thanks for getting back to me on this!

**Comment by bogdanprodan-okta** [comment]
No problem! Sometimes the sequence of the resource creation is essential. That's why terraform came up with `depends_on` field. 
If you don't have any more questions, please close this issue or post them here. Thanks!
